2 d heisenberg

weisongl opened the pull request:
mapping for 2D heisenberg to compute dielectric constant

weisongl committed:
rename heisenberg to be Ising because it's doing ising
create the spin ising package

weisongl committed:
add IPotentialAtomicSecondDerivative
add potentialCalculationFSum
add potentialCalculationPhiSumHeisenberg

weisongl committed:
create heisenberg package

weisongl committed:
add PotentialCalculationPhiSumHeisenberg

weisongl committed:
add potentialCalculationFSUM and potentialCalculationPhiSumHeisenberg

weisongl committed:
add IPotentialAtomicSecondDerivative

weisongl committed:
Merge remote-tracking branch 'origin/rename_spin_ising' into rename_spin_ising

Conflicts:
	etomica-core/src/main/java/etomica/potential/PotentialCalculationPhiSumHeisenberg.java

weisongl committed:
delete useless TODO

weisongl committed:
implement of 2D phisum(secondDerivative sum)

weisongl committed:
delete useless TODO

weisongl committed:
delete useless TODO

weisongl committed:
implement of FSum

weisongl committed:
add variance contribution to dielectric constant

weisongl committed:
edit the torque

weisongl committed:
implement the torque1D

weisongl committed:
correct orientation value, used to be x=y for orientation

weisongl committed:
delete useless comment

weisongl committed:
add a test calculation for the total dipolemoment

weisongl committed:
Rename MeterMappedAveraging, the spell was wrong before "MeterMappedAveriging"

weisongl committed:
replace numberMolecules with nCells, the total number of molecules equals to nCells*nCells

weisongl committed:
delete useless force part, we only use torque here

weisongl committed:
delete useless force part, we only use torque here

akofke committed:
Remove some redundant files

akofke committed:
remove todo

weisongl committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

weisongl committed:
Add commment for Heisenberg.java

weisongl committed:
Add comments for Heisenberg.java

weisongl committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

weisongl committed:
add comment about what this file compute.

weisongl committed:
add comment: intro of what this class computes

weisongl committed:
add comment: intro of what this class computes
remove TODO

akofke commented on etomica-apps/src/main/java/etomica/spin/heisenberg/Heisenberg.java:
```diff
@@ -60,7 +60,11 @@
     
     /**
-     * 
+     * 2D heisenberg model in square lattice.
```
Javadoc requires a separate `@param` annotation for each parameter. So it would be like this:

```java
/**
 * @param _space 
 * @param nCells the total number of atoms
 * @param temperature 
 * @param interactionS the J in heisenberg energy function U = J*Cos(theta1-theta2)
 * ... and so on
 */
```

akofke commented on etomica-apps/src/main/java/etomica/spin/heisenberg/Heisenberg.java:
```diff
@@ -60,7 +60,11 @@
+	 * interactionStrength is the J in heisenberg energy function: U = J*Cos(theta1-theta2)
+	 * dipoleMagnitude is the strength of heisenberg dipole.
      */
     public Heisenberg(Space _space, int nCells, double temperature, double interactionS, double dipoleMagnitude) {
```
Does it need a space param? Isn't it known to be 2D?

akofke commented on etomica-apps/src/main/java/etomica/spin/heisenberg/P1MagneticField.java:
```diff
@@ -14,38 +14,41 @@
 
     /**
-     * @param space
+     * @param space,dipoleMagnitude
```
These should also be separate `@param`s

weisongl committed:
edit params of Heisenberg

weisongl committed:
edit params of Heisenberg
still need to remove the space

weisongl committed:
add params for public method

weisongl committed:
remove _space

weisongl committed:
correct the sign of torque and secondDerivative

weisongl committed:
Add a debug test

weisongl committed:
remove pcforce which is not used

weisongl committed:
remove pcforce which is not used

weisongl committed:
remove TODOs

weisongl committed:
remove pcforce which is not used
Add torqueSum
comment out the variance contribution since it's statistic zero

weisongl committed:
correct the sign of secondDerivative

weisongl committed:
remove unused variables

weisongl committed:
Rename JCoc to be Cos since dot product only returns cos

weisongl committed:
add J in secondDerivativeSum
remove unused agentManager

weisongl committed:
simplify phi Sum

weisongl committed:
add coupling in secondDerivative

weisongl committed:
prove that the Variance is zero (it's commented out)

weisongl committed:
test for lower temperature, for example 10K
reduce the number of molecules to 400

weisongl committed:
move equilibration before setting up blocks

weisongl committed:
add a heisenberg3D package, inforcing atom in x and y 2D space and test the result with our heisenberg2D results

weisongl committed:
Add randomNumber generater

weisongl committed:
remove unused params

weisongl committed:
remove unused import
test difference between MeterSpinMSquare and Meterdipole1cite, turns out to be the same for now

weisongl committed:
revise Meter to adjust for the special 3D case(with dipole no z component or dipoles only in x,y plan)

weisongl committed:
rewrite torque back to be a vector
secondDerivative back to be a matrix

weisongl committed:
remove unused method setDipoleSource

weisongl committed:
add J

weisongl committed:
rewrite phisum

weisongl committed:
Merge branch '2DHeisenberg' of C:\Users\weisong\IdeaProjects\etomica with conflicts.

weisongl committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

weisongl committed:
add a heisenberg3D package, inforcing atom in x and y 2D space and test the result with our heisenberg2D results

weisongl committed:
add printout for random seed , number of blocks and samples per block

weisongl committed:
add test method to verify whether some part of the expression for AEE is zero or not

weisongl committed:
simplify the code and add a second method to compute secondDerivative

weisongl committed:
Add a few test methods to compare java and mathematica results
test for first and secondDerivative of energy

akofke committed:
Merge branch 'master' into 2DHeisenberg

akofke committed:
Fix breakages caused by merging api refactor

From master branch. Also reformatted and rearranged.

weisongl committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

# Conflicts:
#	etomica-apps/src/main/java/etomica/spin/heisenberg/P2Spin.java
#	etomica-core/src/main/java/etomica/potential/PotentialCalculationPhiSumHeisenberg.java

weisongl committed:
Reduce the block numbers to be 100.
Print out Number of molecules(nCells*nCells), interactionS(coupling strength) and dipoleStrength
Divide the output by number of molecules(nCells*nCells) to get value per molecules.

weisongl committed:
Enable 3D rotation

weisongl committed:
Edit the top comment

weisongl committed:
Delete unused codes

weisongl committed:
Enable 3D calculation

weisongl committed:
Remove final for vector ei and ej

kofke committed:
Don't overwrite central atom with neighbor when iterating down site list

weisongl committed:
delete unused classes

weisongl committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

weisongl committed:
change back to 2D

weisongl committed:
delete unused class

kofke committed:
Compute ei, ej before using them, not after

weisongl committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

weisongl committed:
Add a boolean diffInterval to set different interval for mapped averaging and conventional method.

weisongl committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

weisongl committed:
Implement the interacting heisenberg model at zeroth order expansion.

weisongl committed:
remove final for vector ei and ej

weisongl committed:
Add expression for Ans in x direction(completed 0th term for now).
Add debug code to test express of AEE in zeroth order term(completed constant and Fsum part for now)

weisongl committed:
Add Ans expression and its debug

weisongl committed:
Merge branch 'master' into 2DHeisenberg

Conflicts:
	etomica-apps/src/main/java/etomica/spin/ising/MCMoveSpinFlip.java
	etomica-apps/src/main/java/etomica/spin/ising/MeterSpin.java

weisongl committed:
Use bessel function from specialFunction instead of besselFunction.

weisongl committed:
Solve the problem of besselFunction when its order <=0

weisongl committed:
Solve the problem of besselFunction when its order <=0 && k=0

weisongl committed:
Implement expressions for dAxc1 Axs1 and dAxs1
Choose random seed to be 1 for the purpose of debugging

weisongl committed:
Implemented Ans expression for x direction;

Fix one problem of bessel function when order is big (the problem was that the computation of bessel function would keep going till Infinity with the loop).

weisongl committed:
Implement An expressions in y direction.

Make the random seed back to random.

weisongl committed:
Implement d2A0ns and d3A0ns.
Separate computation of x and y direction into two blocks.

weisongl committed:
Implement expression for x direction.

weisongl committed:
Implement expressions of d2A0ns, d3A0ns and d2A1ns for y direction.

weisongl committed:
Test expressions of A0n dA0n d2A0n d3A0n dA1n d2A1n. The random seed is set to be random. Compare numerical results with Mathematica by transfer Java print out into Mathematica code.  100% sure about the accuracy of these expressions.

weisongl committed:
Save value of the Ans expression into array for implementation of JEE-JEJE and UEE.

weisongl committed:
Implemented expression for JEE-JEJE, UEE and Var<JE-UE>.

weisongl committed:
Change the randomSeed to be 1 for debugging expression of AEE.

weisongl committed:
Double check the expression for psix1 etc.

weisongl committed:
Compared the numerical value for psins, vEns, vEEns, JEns, JEEns, UEns, JEEMUEE and JEMUE with mathematica

weisongl committed:
Compared for <Sum(mui+muJ)^2> and sum(AEEij)_nbrs.

weisongl committed:
Seperate contribution of JEMUE into JEMUEx and JUMUEy parts.

weisongl committed:
Correction for Ayc1 expression

weisongl committed:
Corrected expression for JEMUEy
Rename PotentialCalculationPhiSumHeisenberg to be PotentialCalculationPhiSum

weisongl committed:
Add TODOs for future debug

weisongl committed:
Randomize the initial orientation
Rename MeterMappedAveragingTest to be MeterPairSum
Change nMax to be 10
Use atan2 instead of acos to get angle ranging from -pi to pi

weisongl committed:
Tested the pair force summation.

weisongl committed:
removed some debug code

weisongl committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

# Conflicts:
#	etomica-apps/src/main/java/etomica/spin/heisenberg_interacting/heisenberg/HeisenbergTest.java
#	etomica-apps/src/main/java/etomica/spin/heisenberg_interacting/heisenberg/PotentialCalculationHeisenberg.java

weisongl committed:
Finish debug with force.
Going to test how each contribution change with system size.

weisongl committed:
Remove average of JEMUE since it's statistically zero
Combine zerosum methods in potentialCalculationHeisenberg

weisongl committed:
Remove average of JEMUE since it's statistically zero
Combine zerosum methods in potentialCalculationHeisenberg
Separate contribution of AEE to analysis each one.

weisongl committed:
Chose randomSeed to be one.

Compared java with mathematica auto expressions.

apoorvapurohit committed:
changes to 2D dielectric code

weisongl committed:
Compute AEE output by combining -aveJEEMJEJE + aveUEE + aveJEMUESquare into one average sum0
Print out error, difficulty and correlation of AEE output
Reduced nMax in Meter to 10, since nMax=10 has good results with more efficiency
reformate the code
commented out some unused print out.

Implementing 3 pairs heisenberg system and testing the results.

weisongl committed:
Made randomSeed to be one;
Change the temperature to 5k;
Implemented <M^2> = <M^2>_Ideal + Sum_nbr( <M^2>_J - <M^2>_J0);
Add a field numberMolecules for 3 pairs;
Delete isGraphic field
Add IAtomOriented and PotentialCalculationPhiSumHeisenberg to calculated ideal total dipole moment.
Add AEEJ0, JEMUExIdeal and JEMUEyIdeal

weisongl committed:
Print out results of Conventional, IdealMapping , Mapping , JEEMJEJE and UEE

kofke committed:
Progress in implementing pairwise mapping (still not correct yet though)

Will need attention to efficiency once errors are removed

weisongl committed:
Add code for comparing ideal dipole and J=0 case.
Print out JEMUEx JEMUEy
disable rotation of one atom for 3pairs
simplify AEEJ0 expression.

weisongl committed:
Compare AEE direct approach, agree with mathematica but not convention, proved that previous algorithm was wrong. The next step to so test the new approach <M^2>_1+ Sum_interact(<M^2>_J-<M^2>_J0), parts by parts with mathematica

weisongl committed:
Removed unused codes
Add codes to compare with Mathematica analytical results

weisongl committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

Conflicts:
	etomica-apps/src/main/java/etomica/spin/heisenberg_interacting/heisenberg/Heisenberg3Pair.java
	etomica-apps/src/main/java/etomica/spin/heisenberg_interacting/heisenberg/HeisenbergPair.java
	etomica-apps/src/main/java/etomica/spin/heisenberg_interacting/heisenberg/MeterMappedAveraging3Pair.java

weisongl committed:
Print out JEMUE and JEMUESquare
Print out AEEDirect(-JEEMJE+UEE) and AEEDirect(J=0)
Added JEMUESquare in meter
Added debug code in PhisumHeisenberg to test the correctness of the formulas

weisongl committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

weisongl committed:
Removed JEMUE since its average is zero.
Print out different parts of ideal mapping.
Enable 3 pair interacting.
Added secondDerivativeSumIdeal for each pair.

weisongl committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

weisongl committed:
Commented out unused pair meter, which is inefficient.
Correct the variance part of JEMUE.

weisongl committed:
print out dipole moment in x and y direction

weisongl committed:
Summation vE of all neighbours as vE for each atom by:
1.Add field in MoleculeAgent for the terms need to summation first like vE vEE dvEEdtheta etc.
2.Created a PotentialCalculation for the terms need to summation first like vE vEE dvEEdtheta etc.
3. Need to create another PotentialCalculation for pairs as well.

weisongl committed:
Add PotentialCalculationPair to calculate pair contribution.

weisongl committed:
Add meter for VSum: Loop every atom for pair and individual summation.
Apply new meter in Heisenberg for comparison with previous method.

weisongl committed:
Apply VSum in only one pair case, added new meter for pair

weisongl committed:
Corrected a small bug in meter.

weisongl committed:
add debug code to compare result with mathematica analytical result
reset leafAgentManager

weisongl committed:
printed out pvx terms to compare with mathematica,
found a bug: pvx20 terms should exclude n=0 terms since they are zero.

weisongl committed:
Print out JEMUESquare.
Seperated contribution of AEEVSum for comparison.
UEEself terms seems need to be zero and JEEMJEJE term seems to be double than correct answer
corrected sign of dpy1Rpy0dt1 and dpy1Rpy0dt2

weisongl committed:
The vSum method is working, comparing to conventional and one pair mapping case.
Separated UEE and JEEMJEJE into self term(vE is summation from all pairs ) and pair term(vE is only from pair)
corrected two bug in the expression of JEEMJEJE and UEE(JEEMJEJE the notation changed and UEE pair changed to negative)

weisongl committed:
Rearranged the output so that one could directly do the comparison.

weisongl committed:
test for vSum in two pair
reduced nMax to 3 for quick test purpose.

weisongl committed:
Created MeterMappedAveragingVSum3Pair
Applied vsum method to two pairs(three atoms) system using MeterMappedAveragingVSum3Pair
Seperate PotentialCalculationMoleculeAgentSum into pair only and calculate self term only.

weisongl committed:
Perform pair vSum after self terms done calculation.
JEEMJEJE passed the test. UEEpair terms passed the test, now need to compare UEE self terms and JEMUE terms.

weisongl committed:
Apply vSum method to all pairs heisenberg model(2D square), the result is accurate but way less accurate than previous pair mapping or ideal mapping. May need to double check.

weisongl committed:
Created potential calculation for vSum minus ideal

weisongl committed:
Created potential calculation for vSum minus ideal for 3 atoms and 9 atoms system.
VSum minus ideal is working.
Now do the 3 atoms, 2 pair case

weisongl committed:
debug for 2paris/3atoms vSumMinusIdeal Mapping.
Swith the notation sequence to be 01/12 for better comparison with mathematica
Find the value of d2vEx2 not match. Should continue to dig

weisongl committed:
Implemented vSumMinusIdeal in the 3 atoms system, which improve the accuracy for higher temperature( for bt*J >2.0).
Commented out debug codes.And removed unused fields
Corrected functions of ideal contribution in potential calculation and meter.

weisongl committed:
add booleans doIdeal, doPair,doVSum,doVSumMI for separate run to compute difficulties of each method.

weisongl committed:
reduced nMax to be 5 since results barely change for nMax = 5 or 10 but the time could reduce by half.
Optimize the calculation in potential calculation. For example create field like I0bJ since it's a constant. Other example would be assign cost1 = ei.getX(0) to avoid trig calculation and replace cos((n-2)*t1) to be cosnt1*cost1+sinnt1*sint1 etc.
Combine x and y direction for loop calculation into one.

weisongl committed:
Introduced the field nMax.
Reduced the

weisongl committed:
Added nMax for heisenbergPair and heisenberg3Pair

weisongl committed:
used bJ instead of J for the meter.
Implemented the expression of Ans in the meter.
Comment out t1 and t2 in potential calculation since we used ei.getX(0) to replace cost1 etc
Precomputed sin(n*t2) to void repeated calculation.
Removed ideal contribution in potential calculation since this part was moved to meter to avoid repeated calculation.

weisongl committed:
Cleaned out most of the debug or unused code.
Computed sin(n*theta) and cos(n*theta) in the meter(input to moleculeAgent) to saved 8 times more calculation in potential calculation.

weisongl committed:
passed value of sinntheta in agent to the meter.

weisongl committed:
Print out net dipole moment in x and y direction.

ajschult committed:
Make orientation line more useful, other random cleanup

ajschult committed:
Set positions for sites, add code for graphics

weisongl committed:
Add conventional/mapping way to compute the free energy of the system.
Computed the variance of free energy in conventional/mapping way

weisongl committed:
add new meter and potentialCalculation for only free energy by mapping

ajschult committed:
Add cluster move for spins

ajschult committed:
Use new cluster move, add temperature slider

ajschult committed:
cleanup

ajschult committed:
tell iterator do go up and down

weisongl committed:
Added meter for heat capacity.
Moved the zeroSum of MoleculeAgent from potentialCalculation to MoleculeAgent.

weisongl committed:
Add a boolean doGraphic to decide run graphic or not.
Moved some printout closer to when the value are computed to avoid confusion.
Computed Free energy and its variance.

weisongl committed:
Print out CV directly for comparison

kofke committed:
Correction of both coding and formula errors in calculation of mapped-averaging Cv and its uncertainty.

Results not yet matching literature.

kofke committed:
Experimenting with different ways to visualize spins

kofke committed:
Merge branch '2DHeisenberg' of github.com:etomica/etomica into 2DHeisenberg

weisongl committed:
Fixed a tiny output of CV by dividing temperaure because T*CV = beta*var[u]

kofke committed:
Mapped-averaging heat-capacity meter now gives correct results

kofke committed:
Merge remote-tracking branch 'origin/2DHeisenberg' into 2DHeisenberg

kofke committed:
Also need correction to main file that runs the calculation

weisongl committed:
Implement mapped average correlation meter.

weisongl committed:
Apply MeterMappingAveragingCorrelation in heisenberg and print out results.
Add notes to separate different methods and output

weisongl committed:
add conventional output for correlation
compute average of correlation instead of sum
Use custom agent to correct torque to be Vector1D

weisongl committed:
doubled he sampleInterval
add field formula to choose formula 0 1 or 2

weisongl committed:
Add row and col number to the correlation output

weisongl committed:
Correct the output sign for formula 1 and 2.

weisongl committed:
Add second derivative term for x[0] in formula 2. It's working Now!!

ajschult committed:
start of mean-field mapping data collection

kofke committed:
New class to support effective-field mapped averaging

ajschult committed:
get the torque as well

ajschult committed:
Implement net dipole moment and correlation with mean field mapping

ajschult committed:
Need meter for dipole moment too

ajschult committed:
maybe less wrong mapping velocities

kofke committed:
Correct a sign error

kofke committed:
Need to difference integrals, not p * integrals

kofke committed:
Handle correlation for neighbors correctly (apparently)

This is for effective-field mapped average, 2nd-derivative method

Does not consider possible contributions from derivatives of effective
field w.r.t. neighbor orientations

ajschult committed:
Add energy data collection with mean field mapping

kofke committed:
Fix incorrect division by T

ajschult committed:
Use iterator properly

kofke committed:
Implement effective-field mapped average for the energy

Add some more information to output from Heisenberg

kofke committed:
Add new function to support mapped-average calculation of heat capacity

ajschult committed:
add mean field Cv

kofke committed:
Initial implementation of ij contribution to mapped-averaging heat capacity

Work in progress.  Does not yet produce correct values.

kofke committed:
Squash some bugs in calculation of heat capacity with effective-field mapped averaging

Appears to give correct result, but more testing is needed

kofke committed:
Implement ij contribution to spin-spin correlation for effective-field mapped averaging

Needed only for neighbor spins

kofke committed:
Development in progress for AEE via mean-field mapping (not complete)

ajschult committed:
clean up spin package
drop simple 2D and 3D Heisgenberg packages
drop magnetic field class where not used
